wasm/jit/jit_amd64.go

```diff
@@ -693,25 +702,140 @@ func (c *amd64Compiler) compileCall(o *wazeroir.OperationCall) error {
 	}
 
 	target := c.f.ModuleInstance.Functions[o.FunctionIndex]
-	if target.HostFunction != nil {
-		index := c.eng.compiledHostFunctionIndex[target]
-		if err := c.callHostFunctionFromConstIndex(index); err != nil {
-			return err
-		}
-	} else {
-		index := c.eng.compiledWasmFunctionIndex[target]
-		if err := c.callFunctionFromConstIndex(index); err != nil {
-			return err
+	if err := c.makeFunctionCallFromConsts(jitCallStatusCodeCallFunction, target.Address); err != nil {
+		return err
+	}
+
+	// We consumed the function parameters from the stack after call.
+	for i := 0; i < len(target.FunctionType.Type.Params); i++ {
+		c.locationStack.pop()
+	}
+
+	// Also, the function results were pushed by the call.
+	for _, t := range target.FunctionType.Type.Results {
+		loc := c.locationStack.pushValueOnStack()
+		switch t {
+		case wasm.ValueTypeI32, wasm.ValueTypeI64:
+			loc.setRegisterType(generalPurposeRegisterTypeInt)
+		case wasm.ValueTypeF32, wasm.ValueTypeF64:
+			loc.setRegisterType(generalPurposeRegisterTypeFloat)
 		}
 	}
+	return nil
+}
+
+const (
+	tableElementFunctionAddressOffest = 0
+	tableElementTypeIDOffest          = 8
+)
+
+// compileCallIndirect adds instructions to perform call_indirect operation.
+// This consumes the one value from the top of stack (called "offset"),
+// and make a function call against the function whose function address equals "table[offset]".
+// This is called indirect function call in the sense that the target function is indirectly
+// determined by the current state (top value) of the stack.
+// Therefore, two checks are performed at runtime before entering the target function:
+// 1) If "offset" exceeds the length of table, "out of bounds table access" is raised.
```

Comment:
probably better to use symbols vs text as they can reference the doc on them
```suggestion
// 1) If "offset" exceeds the length of table, "out of bounds table access" states (jitCallStatusCodeTableOutOfBounds) is returned.
```